app/scripts/services/fight.js

```diff
@@ -113,149 +52,189 @@ angular.module("rpgApp").service("FightEngine", ["CharServ", "AdversariesDB", "I
     // regain 1 mana pro round
     CharServ.manaRegen();
 
-
-    // player actions
-
-    // attack roll
-    var playerAtt = actionRoll();
-
-    if (isCriticalSuccess(playerAtt, $scope.player.weapon.critical[0])) {
-      // critical success hits automatically with improved damages
-      playerAtt += $scope.player.stats.hitBonus;
-      var playerCritDmg = (rollDamages($scope.player.weapon.damages) + $scope.player.attribute.strength) * $scope.player.weapon.critical[1];
-      $scope.mob.life -= playerCritDmg;
-
-      $scope.messages.push({
-        text: "You attack: " + playerAtt + "   -- CRITICAL HIT!",
-        type: "attack"
-      });
-
-      $scope.messages.push({
-        text: "You hit the enemy and inflict " + playerCritDmg + " damages.",
-        type: "damagesToMob",
-        dmg: playerCritDmg
-      });
-
-    } else if (isCriticalFailure(playerAtt)) {
-      // critical failure misses always and cause bad outcome
-      playerAtt += $scope.player.stats.hitBonus;
-      $scope.player.stats.life -= 2;
-
-      $scope.messages.push({
-        text: "You attack: " + playerAtt + "   -- CRITICAL FAILURE!",
-        type: "attack"
-      });
-
-      $scope.messages.push({
-        text: "You hit yourself for " + 2 + " damages.",
-        type: "damagesToPlayer",
-        dmg: 2
-      });
-    } else {
-      playerAtt += $scope.player.stats.hitBonus;
-
-      if (isSuccess(playerAtt, $scope.mob.defence)) {
-        // do some damages if the attack hit
-        var playerDmg = rollDamages($scope.player.weapon.damages) + $scope.player.attribute.strength;
-        $scope.mob.life -= playerDmg;
+    var player = opponents.player;
+    var mob = opponents.mob;
+
+    // 1st: player attacks!
+    var attackFromPlayer = player.attack();
```

Comment:
Start from here to review the code. Jump inside the `attack` implementation, and make your way through the code. Ask me any questions that come out.